### PR TITLE
reef: ceph.spec.in: add support for openEuler OS

### PR DESCRIPTION
**Pull request by Svelar**
backport tracker: https://tracker.ceph.com/issues/65032

---

backport of https://github.com/ceph/ceph/pull/50301
parent tracker: https://tracker.ceph.com/issues/65029

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/main/src/script/ceph-backport.sh

### CODE REVIEW

**Review by idryomov — COMMENTED** [review]
Where did "[doc: fix get-packages.rst typos](https://github.com/ceph/ceph/pull/56361/commits/95622fb5fca620777cf5e9c9fa0d99c821e2f31a)" commit come from?  I think it should be submitted in a separate PR against main.  Once it merges, the docs team would backport it.

**Comment by Svelar** [comment]
> Where did "[doc: fix get-packages.rst typos](https://github.com/ceph/ceph/pull/56361/commits/95622fb5fca620777cf5e9c9fa0d99c821e2f31a)" commit come from? I think it should be submitted in a separate PR against main. Once it merges, the docs team would backport it.

Done. Same as #56366

**Comment by idryomov** [comment]
> Same as #56366

See my comment there.

**Review by idryomov — APPROVED** [review]

**Review by anthonyeleven — APPROVED** [review]

**Comment by Svelar** [comment]
https://pulpito.ceph.com/svelar-2024-03-27_06:35:39-rados-wip-svelar-testing-reef-2024-03-26-0716-distro-default-smithi/

Seems like unrelated failed and dead jobs

**Comment by Svelar** [comment]
> https://pulpito.ceph.com/svelar-2024-03-27_06:35:39-rados-wip-svelar-testing-reef-2024-03-26-0716-distro-default-smithi/
> 
> Seems like unrelated failed and dead jobs

Would merge this one if no objection occurs in two days.

**Comment by tchaikov** [comment]
@Svelar Yuri has added a label noting that your PR has been included in one of the test batches. i'd suggest wait for his test result before merging this change.

**Comment by Svelar** [comment]
> @Svelar Yuri added a label noting that your PR has been included in one of the test batches. i'd suggest wait for his test result before merging this change.

Yeh, will wait until then.

**Comment by ljflores** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/REEF#httpstrackercephcomissues65202

**Pull request merged**
